File: Java/Programs/Clothing.java
/**
 * Clothing child of product
 */
class Clothing extends Product{
    // atribut produk protected agar dapat digunakan oleh keturunannya & no acces langsung(kalau beda package)
    protected String size; // cloth size
    protected String material; // cloth mat
    protected String gender; //  cloth gender
    // contruct kosong
    protected Clothing(){
    }
    //construct with value
    protected Clothing(String id, String name, String brands, double price,String size, String material, String gender){
        super(id,name, brands,price); // mengisi variabel yang dipunya orang tuanya
        this.size = size;
        this.material =material;
        this.gender = gender;
    } 
    
    // set void
    void set_size(String size){
        this.size = size;
    }
    void set_material(String material){
        this.material = material;
    }
    void set_gender(String gender){
        this.gender = gender;
    }

    // get method
    String get_size(String size){
        return this.size;
    }
    String get_material(String material){
        return this.material;
    }
    String get_gender(String gender){
        return this.gender;
    }
    
    
}

File: Python/Programs/Clothing.py
# Import kelas dasar Product
from Product import Product

# Derived class representing a clothing product, inheriting from the Product base class
class Clothing(Product):
    # Constructor
    def __init__(self, id="", name="", brands="", price=0, size="", material="", gender=""):
        # Call the constructor of the base class (Product)
        super().__init__(id, name, brands, price)
        # Initialize additional attributes with default or provided values
        self.size = size
        self.material = material
        self.gender = gender

    # Setter methods

    # Set the clothing size
    def set_size(self, size):
        self.size = size

    # Set the clothing material
    def set_material(self, material):
        self.material = material

    # Set the clothing gender
    def set_gender(self, gender):
        self.gender = gender

    # Getter methods

    # Get the clothing size
    def get_size(self):
        return self.size

    # Get the clothing material
    def get_material(self):
        return self.material

    # Get the clothing gender
    def get_gender(self):
        return self.gender
